sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -397,6 +397,10 @@ RT::PiProgram ProgramManager::getBuiltPIProgram(OSModuleHandle M,
               Img.getLinkOptions(), PiDevices,
               ContextImpl->getCachedLibPrograms(), DeviceLibReqMask);
 
+    {
```

Comment:
is this  change related to the feature being implemented?

Reply:
Yes, it is. We need to register the built program as well as the created one, since those might be different due to linking with our device library images.